frontend/src/pages/_middleware.ts

```diff
@@ -3,15 +3,20 @@ import { NextResponse } from "next/server";
 
 import { DEFAULT_LOCALE, VALID_LOCALES } from "~/locale/languages";
 
+const PUBLIC_FILE = /\.(.*)$/;
+
 export function middleware(req: NextRequest, _ev: NextFetchEvent) {
-  const { pathname } = req.nextUrl;
+  const shouldHandleLocale =
+    !PUBLIC_FILE.test(req.nextUrl.pathname) &&
+    !req.nextUrl.pathname.includes("/api/") &&
+    req.nextUrl.locale === "default";
   const locale = getLocale(req.cookies.pyconLocale);
 
-  if (pathname === "/") {
-    return NextResponse.redirect(`/${locale}`);
-  }
-
-  return NextResponse.next();
+  return shouldHandleLocale
+    ? NextResponse.redirect(
+        `/${locale}${req.nextUrl.pathname.replace("/default", "")}`,
+      )
+    : undefined;
```

Comment:
very nit: I'd use an if instead of a ternary operator here, especially because we are returning undefined in one case :)

Reply:
👍